src/diffusers/pipelines/deprecated/versatile_diffusion/modeling_text_unet.py

```diff
@@ -1275,13 +1275,16 @@ def forward(
             image_embeds = added_cond_kwargs.get("image_embeds")
             encoder_hidden_states = self.encoder_hid_proj(image_embeds)
         elif self.encoder_hid_proj is not None and self.config.encoder_hid_dim_type == "ip_image_proj":
-            if "image_embeds" not in added_cond_kwargs:
+            if cross_attention_kwargs is not None and cross_attention_kwargs.get("ip_hidden_states", None) is not None:
+                ip_hidden_states = cross_attention_kwargs.pop("ip_hidden_states")
```

Comment:
if you pop the `ip_hidden_states`, the next step throws the `ValueError` since there's no `image_embeds` and no `ip_hidden_states`. If you don't do it though, there's a heavy spam in the console, this is an expected issue with not going with the `kwargs` route.

Reply:
umm i see. it make sense!
do you think we should pass it to `ip_adapter_image_embeds` and decide whether it has been projected or not based on the shape?

Reply:
yeah, that's a good solution, it works for me.